jws.go

```diff
@@ -46,6 +46,7 @@ type SignatureConfig struct {
 type SignerConfig struct {
 	Alg                string   `json:"alg"`
 	KeyID              string   `json:"kid"`
+	Type               string   `json:"typ"`
```

Comment:
Why are you adding this new field?

Reply:
to unify the output across different outputs if you think it is not necessary it is totally ok for me to remove it. it is my first commit from many to come I hope so I am open to discussions for sure

Reply:
This struct contains the configuration for the signer middleware. By adding this new field, we could let the users select the type of token to generate. For that, the `SignerOptions` should be created after checking this new field (and the `typ` header should contain whatever the config says. Something like this:
```
if signerCfg.Type != "" {
opts.ExtraHeaders[jose.HeaderKey("typ")] = signerCfg.Type
}
```